crates/orchestrator/src/jobs/mod.rs

```diff
@@ -183,38 +211,42 @@ fn get_job_handler(job_type: &JobType) -> Box<dyn Job> {
     }
 }
 
-async fn get_job(id: Uuid) -> Result<JobItem> {
+async fn get_job(id: Uuid) -> Result<JobItem, JobError> {
     let config = config().await;
     let job = config.database().get_job_by_id(id).await?;
     match job {
         Some(job) => Ok(job),
-        None => {
-            log::error!("Failed to find job with id {:?}", id);
-            Err(eyre!("Failed to process job with id {:?}", id))
-        }
+        None => Err(JobError::JobNotFound { id }),
     }
 }
 
-fn increment_key_in_metadata(metadata: &HashMap<String, String>, key: &str) -> Result<HashMap<String, String>> {
+fn increment_key_in_metadata(
+    metadata: &HashMap<String, String>,
+    key: &str,
+) -> Result<HashMap<String, String>, JobError> {
     let mut new_metadata = metadata.clone();
     let attempt = get_u64_from_metadata(metadata, key)?;
     let incremented_value = attempt.checked_add(1);
     if incremented_value.is_none() {
```

Comment:
```rs
let Some(incremented_value) = incremented_value else {
  return Err(JobError::KeyOutOfBounds { key: key.to_string() }) }
};
``` to avoid the unwrap just after

Reply:
or in this case just, `incremented_value.ok_or_else(|| JobError::KeyOutOfBounds { key: key.to_string() })`

Reply:
Good catch !
Applied.